cupyx/jit/_builtin_funcs.py

```diff
@@ -28,6 +53,25 @@ def call(self, env, *args, **kwargs):
             raise TypeError(
                 f'range expected at most 3 argument, got {len(args)}')
 
+        unroll = kwargs.get('unroll')
+        if unroll is not None:
+            if not all(isinstance(x, Constant)
+                       for x in (start, stop, step, unroll)):
+                raise TypeError(
+                    'loop unrolling requires constant start, stop, step and '
+                    'unroll value')
+            unroll = unroll.obj
+            if not (isinstance(unroll, int) or isinstance(unroll, bool)):
+                raise TypeError(
+                    'unroll value expected to be of type int, '
+                    f'got {type(unroll).__name__}')
+            if unroll is False:
+                unroll = 1
```

Comment:
I am not sure how useful it is to set `#pragma unroll(1)`, at least I haven't encountered such need in real life yet. Let's have @asi1024 make the final call 🙂

Reply:
@kmaehashi [mentions in the first bullet](https://github.com/cupy/cupy/pull/5343#issuecomment-862842340) that the CUDA doc states:
> By default, the compiler unrolls small loops with a known trip count.

I agree it's not a common use case, but `#pragma unroll(1)` explicitly disables unrolling, which is necessary for users to take full control of the unrolling behavior.

Reply:
The users can control it by specifying `unroll=1` instead.

Reply:
How about mapping `False` to `#pragma nounroll`?
https://github.com/NVlabs/xmp/blob/0052dbb206ef0cb2b47b0fe5d61f54e5cd2fa73c/src/include/digitmp/mul.h#L42

Reply:
I prefer using `"always"`, `"default"`, `"never"` literals instead of `True`, `None` and `False`.

Reply:
@asi1024 As `unroll` is intended for users understanding `#pragma unroll` behavior, I think these mappings are enough clear and straightforward, so +1 not to introduce our own semantics on the argument.
* `unroll=True`: `#pragma unroll`
* `unroll=False`:  `#pragma nounroll`
* `unroll=3`:  `#pragma unroll(3)`
* not specified (or `unroll=None`): `` (no pragma)
* others: error

Reply:
Sorry, I was being silly. @eternalphane was right that `#pragma unroll(1)` is the correct directive to disable unrolling. This is documented in the Programming Guide. 

`#pragma nounroll` has no effect (`nounroll` should be done at the PTX level, not the C++ level). 

We can do a little experiment with https://godbolt.org/z/soanE1hhv by observing the output PTX length for different pragma setups.